app/web/src/features/search/FilterDialog.tsx

```diff
@@ -77,88 +116,136 @@ export default function FilterDialog({
   });
 
   //prevent defaultValue changing
-  const defaultValues = useRef({
-    location:
-      searchFilters.active.location &&
-      searchFilters.active.lng &&
-      searchFilters.active.lat
-        ? {
-            name: searchFilters.active.location,
-            simplifiedName: searchFilters.active.location,
-            location: new LngLat(
-              searchFilters.active.lng,
-              searchFilters.active.lat
-            ),
-          }
-        : undefined,
-    lastActive: lastActiveOptions.find(
-      (o) => o.value === searchFilters.active.lastActive
-    ),
-    hostingStatusOptions: searchFilters.active.hostingStatusOptions,
-    numGuests: searchFilters.active.numGuests,
-  }).current;
+  const defaultValues = useMemo(
+    () => ({
+      location:
+        searchFilters.active.location &&
+        searchFilters.active.lng &&
+        searchFilters.active.lat
+          ? {
+              name: searchFilters.active.location,
+              simplifiedName: searchFilters.active.location,
+              location: new LngLat(
+                searchFilters.active.lng,
+                searchFilters.active.lat
+              ),
+            }
+          : ("" as const),
+      keywords: searchFilters.active.query ?? "",
+      lastActive:
+        lastActiveOptions.find(
+          (o) => o.value === searchFilters.active.lastActive
+        ) ?? null,
+      hostingStatusOptions: searchFilters.active.hostingStatusOptions,
+      numGuests: searchFilters.active.numGuests ?? "",
+    }),
+    //we specifically want abnormal behaviour of not updating default values
+    //eslint-disable-next-line react-hooks/exhaustive-deps
+    [isOpen]
+  );
```

Comment:
After reading this about `undefined` being an invalid value for `defaultValue`](https://github.com/react-hook-form/react-hook-form/discussions/2797) (e.g. in `location` before you changed to `"" as const`), I think this `useMemo` (or previously `useRef`) might be unnecessary and was really a workaround because we didn't use the library correctly? I had a play around locally by removing the `useMemo` and I didn't see the warning anymore.